src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -42,6 +52,12 @@ public static AccountID encodeToXDRAccountId(String data) {
         return accountID;
     }
 
+    public static Optional<VersionByte> decodedVersionByte(String data) {
+        byte[] decoded = StrKey.base32Encoding.decode(java.nio.CharBuffer.wrap(data.toCharArray()));
+        byte decodedVersionByte = decoded[0];
+        return VersionByte.findByValue(decodedVersionByte);
```

Comment:
The `Optional` usage in `findByValue` is good, because it is a small utility function and using Optional makes it easier to compose into different situations. Also the name findX returning an absent is easily expected and the two go together well.

For this function, `decodeVersionByte`, I think it is more appropriate to turn the missing result from `findByValue` into a `FormatException` in the same way as the other decode functions do, because given valid input the decode process will always result in a known version byte.

```suggestion
        VersionByte versionByte = VersionByte.findByValue(decodedVersionByte);
        if (!versionByte.isPresent()) {
                throw new FormatException("Version byte is invalid");
        }
        return versionByte.get();
```